src/Translation_unit.ml

```diff
@@ -365,23 +346,14 @@ let format xunit (conf : Conf.t) ?output_file ~input_name ~source ~parsed ()
             (* All good, continue *)
             print_check ~i:(i + 1) ~conf t_new ~source:fmted
   in
-  let result =
-    if conf.disable then Ok source
-    else
-      match (parsed : ('a with_comments, exn) Result.t) with
-      | Error exn -> Error (Invalid_source {exn})
-      | Ok t -> (
-        try print_check ~i:1 ~conf t ~source with
-        | Sys_error msg -> Error (User_error msg)
-        | exn -> Error (Ocamlformat_bug {exn}) )
-  in
-  ( match result with
-  | Ok _ -> ()
-  | Error e -> print_error conf Format.err_formatter input_name e ) ;
-  result
+  match parsed with
+  | Error exn -> Error (Invalid_source {exn})
```

Comment:
This should be checked in `parse_and_format` below.
Does not make sense to have a `result` here.

Reply:
indeed, done